app/components/autocomplete/channel_mention/channel_mention.tsx

```diff
@@ -306,21 +274,41 @@ const ChannelMention = ({
         }
 
         setNoResultsTerm(null);
-        runSearch(serverUrl, matchTerm);
-    }, [matchTerm]);
+        setLoading(true);
+        runSearch(serverUrl, matchTerm, teamId);
+    }, [matchTerm, teamId]);
+
+    const channels = useMemo(() => {
+        const ids = new Set(localChannels.map((c) => c.id));
+        return [...localChannels, ...remoteChannels.filter((c) => !ids.has(c.id))].sort((a, b) => {
+            const aDisplay = 'display_name' in a ? a.display_name : a.displayName;
```

Comment:
I think there is a sorting function for this already created, does that one qork here?